Intersect.Client/Entities/Player.cs

```diff
@@ -799,46 +840,79 @@ public void TryWithdrawItem(int bankSlotIndex, int inventorySlotIndex = -1)
                 return;
             }
 
-            //Permission Check
-            if (Globals.GuildBank)
+            // Permission Check for Guild Bank
+            if (Globals.GuildBank && !IsGuildBankWithdrawAllowed())
             {
-                var rank = Globals.Me.GuildRank;
-                if (string.IsNullOrWhiteSpace(Globals.Me.Guild) || (!rank.Permissions.BankRetrieve && Globals.Me.Rank != 0))
-                {
-                    ChatboxMsg.AddMessage(new ChatboxMsg(Strings.Guilds.NotAllowedWithdraw.ToString(Globals.Me.Guild), CustomColors.Alerts.Error, ChatMessageType.Bank));
-                    return;
-                }
+                ChatboxMsg.AddMessage(new ChatboxMsg(Strings.Guilds.NotAllowedWithdraw.ToString(Globals.Me.Guild), CustomColors.Alerts.Error, ChatMessageType.Bank));
+                return;
             }
 
-            var quantity = bankSlot.Quantity;
-            var maxQuantity = quantity;
+            var itemQuantityInBank = bankSlot.Quantity;
+            int availableSpace = FindAvailableInventorySpaceForItem(itemDescriptor.Id);
 
-            if (itemDescriptor.IsStackable)
+            if (availableSpace < 1)
             {
-                maxQuantity = GetQuantityOfItemInBank(itemDescriptor.Id);
+                ChatboxMsg.AddMessage(new ChatboxMsg(Strings.Bank.WithdrawItemNoSpace, CustomColors.Alerts.Error, ChatMessageType.Bank));
+                return;
             }
 
-            if (maxQuantity < 2)
+            if (itemQuantityInBank < 2)
             {
-                PacketSender.SendWithdrawItem(bankSlotIndex, 1, inventorySlotIndex);
+                WithdrawSingleItem(bankSlotIndex, inventorySlotIndex);
                 return;
             }
 
-            int[] userData = new int[2] { bankSlotIndex, inventorySlotIndex };
+            OpenWithdrawItemInputBox(itemDescriptor, bankSlotIndex, inventorySlotIndex, availableSpace);
```

Comment:
Will review after your next commit.

Ideally this should be the exact same as the `TryDeposit`, just swapping the words "bank" and "inventory" to make it go the other direction.

Eventually that logic should also be unified (and when it does it should be unified with the server implementation, we do not need multiple implementations of inventory scanning/transferring in the engine) -- but _don't_ try to make a PR doing that yet.

Reply:
Made sure that both are equally the same in terms of the logic used, will not address the unifying process for now